tests/lib/test_Repo_add.py

```diff
@@ -1,19 +1,18 @@
 from pathlib import Path
+from typing import Collection, Dict, Iterable, List, Union
+
 import pytest
+from onyo import Repo
 
 
-variants = {
-    'str': 'single-file',
-    'Path': Path('single-file'),
-    'list-str': ['single-file'],
-    'list-Path': [Path('single-file')],
-    'set-str': {'single-file'},
-    'set-Path': {Path('single-file')},
-}
-@pytest.mark.parametrize('variant', variants.values(), ids=variants.keys())
-def test_add_args_single_file(repo, variant):
+@pytest.mark.parametrize('variant', [
+    'single-file', Path('single-file'), ['single-file'], [Path('single-file')],
```

Comment:
IMO, this is easier to read if it's one item per line (as it was before). It is more verbose, but much easier to read.